app/plugin_api.go

```diff
@@ -57,6 +63,39 @@ func (api *PluginAPI) LoadPluginConfiguration(dest interface{}) error {
 	}
 }
 
+func (api *PluginAPI) ValidateConfig(config map[string]interface{}) bool {
+	// walk api down to manifest to see if settings meta data is defined
+	// plugin api infratructure might not be completly initialized
+	if config == nil || len(config) < 1 ||
```

Comment:
If the method is made private, I question the need to parameter-validate `api` and `config`, 0/5 seems like we can assume that `api` and `config` are not empty, and maybe even `api.manifest`.

Reply:
I gravitate towards safety giving we are in plugin pre-start stages, so I will post more detailed messages for:
1.  api missing -> log an error that  plugin system is in an invalid state because api is not initialized
2. config -> log an error that  plugin system is in an invalid state because config is not initialized
3. api.Manifest -> log an error that  plugin system is in an invalid state because manifest is not initialized